tests/src/test/java/alluxio/server/ft/journal/raft/EmbeddedJournalIntegrationTestFaultTolerance.java

```diff
@@ -77,6 +87,52 @@ public void failover() throws Exception {
     mCluster.notifySuccess();
   }
 
+  @Test
+  public void syncMetadataFailOver() throws Exception {
+    mCluster = MultiProcessCluster.newBuilder(
+        PortCoordination.EMBEDDED_JOURNAL_FAILOVER_METADATA_SYNC)
+        .setClusterName("EmbeddedJournalFaultTolerance_syncMetadataFailOver")
+        .setNumMasters(NUM_MASTERS)
+        .setNumWorkers(1)
+        .addProperty(PropertyKey.MASTER_JOURNAL_TYPE, JournalType.EMBEDDED)
+        .addProperty(PropertyKey.MASTER_JOURNAL_FLUSH_TIMEOUT_MS, "5min")
+        .addProperty(PropertyKey.MASTER_JOURNAL_CHECKPOINT_PERIOD_ENTRIES, 1000)
+        .addProperty(PropertyKey.MASTER_JOURNAL_LOG_SIZE_BYTES_MAX, "50KB")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_MIN_ELECTION_TIMEOUT, "3s")
+        .addProperty(PropertyKey.MASTER_EMBEDDED_JOURNAL_MAX_ELECTION_TIMEOUT, "6s")
+        .addProperty(PropertyKey.MASTER_STANDBY_HEARTBEAT_INTERVAL, "5s")
+        .build();
+    mCluster.start();
+    mCluster.waitForAllNodesRegistered(30_000);
+    String ufsPath = mFolder.newFolder().getAbsoluteFile().toString();
+    String ufsUri = "file://" + ufsPath;
+    MountPOptions options = MountPOptions.newBuilder().build();
+    FileSystem client = mCluster.getFileSystemClient();
+    AlluxioURI mountPath = new AlluxioURI("/mnt1");
+    client.mount(mountPath, new AlluxioURI(ufsUri), options);
+
+    // create a file outside alluxio
+    String fileName = "someFile";
+    String contents = "contents";
+    try (FileWriter fw = new FileWriter(Paths.get(
+        PathUtils.concatPath(ufsPath, fileName)).toString())) {
+      fw.write(contents);
+    }
+    // sync it with metadata sync
+    assertEquals(contents, IOUtils.toString(client.openFile(
+        mountPath.join(fileName)), Charset.defaultCharset()));
+
+    // restart the cluster
+    mCluster.stopMasters();
```

Comment:
So this is sync then failover, not failover during sync. Do you think failover-during-sync is worth testing? Not necessarily in this PR.

Reply:
Yes its a good idea, especially for UFS journal since it writes different entries to different files which could create interesting inconsistencies.